plugins/modules/source_control/git/git_commit.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020, Federico Olivieri (lvrfrc87@gmail.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: git_commit
+author:
+    - "Federico Olivieri (@Federico87)"
+version_added: "2.10"
+short_description: Perform git and and git commit operations.
+description:
+    - Manage git add and git commit on local git repository.
+options:
+    path:
+        description:
+            - Folder path where .git/ is located.
+        required: true
+        type: path
+    comment:
+        description:
+            - Git commit comment. Same as "git commit -m".
+        type: str
+        required: true
+    add:
+        description:
+            - list of files to be staged. Same as "git add ."
+              Asterisx values not accepted. i.e. "./*" or "*".
+        type: list
+        default: ["."]
+        elements: str
+requirements:
+    - git>=2.10.0 (the command line tool)
+'''
+
+EXAMPLES = '''
+- name: Add and commit 2 files.
+  community.general.git_commit:
+    path: /Users/federicoolivieri/git/git_test_module
+    comment: My amazing backup
+    add: ['test.txt', 'txt.test']
+
+- name: Add all files using default and commit.
+  community.general.git_commit:
+    path: /Users/federicoolivieri/git/git_test_module
+    comment: My amazing backup
+'''
+
+RETURN = '''
+output:
+    description: list of git cli command stdout
+    type: list
+    returned: always
+    sample: [
+        "[master 99830f4] Remove [ test.txt, tax.txt ]\n 4 files changed, 26 insertions(+)..."
+    ]
+'''
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+
+
+def git_commit(module):
+
+    commands = list()
+
+    add = module.params.get('add')
+    path = module.params.get('path')
+    comment = module.params.get('comment')
+
+    if add:
+        commands.append('git -C {path} add {add}'.format(
+            path=path,
+            add=' '.join(add),
+        ))
+
+    if comment:
+        commands.append('git -C {path} commit -m "{comment}"'.format(
```

Comment:
```suggestion
        commands.append('git -C {path} commit -m "{comment}"'.format(
```
Never ever use strings for commands! Use lists. This will also take care of all option quoting problems. (Your current version will break if someone uses `"` in a comment.)

(Note that `module.run_command(cmd, ...)` accepts lists for `cmd`.)

Reply:
Just to make sure I understand properly:

Those strings are appended to a list that is ater passed to `run_command`

```
    for cmd in git_commit(module):
        _rc, output, error = module.run_command(cmd, check_rc=False)
```

Is not that the same?

Reply:
No. Consider `module.run_command('echo "Hello world!"')` vs. `module.run_command(['echo', 'Hello world'])`. Here it is clear what are parameters 1 and 2 of the command, without having to parse it first with shell lexer rules.

Also assume that the commit message is `x " hello " x`. Then your call would result in the arguments `-m`, `"x "`, `hello`, `" x"` (four separate ones) passed to `git`. Using `['-m', 'x " hello " x']` would result in precisely two arguments being passed (with message exactly as provided by user).

Reply:
I see what you mean know. It makes absolutely sense. I will fix that

Reply:
Regarding this bit, I have a problem to make it work.
This is the command: `"cmds": ["git", "add", "1585649737984992100.txt 1585649738063034300.txt"]`

When I run it against testost I get `{"changed": false, "cmd": "add", "msg": "[Errno 2] No such file or directory: b'add': b'add'", "rc": 2}`

If I passe the command as string (using `' '.join()) the command is successful. I had a look around to other modules for [git_config](https://github.com/ansible-collections/community.general/blob/74d83d8e94cab6d72755b3a5b8a476ca49f68d85/plugins/modules/source_control/git_config.py#L249) and I see they use a string for `run_commands`

Reply:
Just because other modules do bad things doesn't mean you should copy their behavior :) Using lists is a lot better than strings for `run_command`.

About that error: where exactly does that come from? Module output is not really helpful to understand what went wrong. What exactly did you pass to `run_command`, and what did `run_command` return?